contracts/MarketCollateralPool.sol

```diff
@@ -190,10 +192,11 @@ contract MarketCollateralPool is Ownable {
         uint256 balanceAfterWithdrawal = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender].subtract(
             withdrawAmount
         );
-
+        uint256 balanceBeforeWithdrawal = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender];
+        int256 changeInBalance = int256(balanceAfterWithdrawal).subtract(int256(balanceBeforeWithdrawal));
```

Comment:
same thing here, shouldn't it always be equal to the withdraw amount?

Reply:
Yes, @pelsasser it should be the same! I already mentioned that in the comments I made while explaining my solution before I started coding, I just did not get a clear direction about it from your replies.